### ISSUE TITLE
Cannot run "docker exec" on Linux on Z

### ISSUE THREAD

**Issue by vivkong** [moby/moby#36467]
**Description**

Cannot create another bash session using "docker exec" on a running container on Ubuntu 16.04 on Linux on IBM Z

**Steps to reproduce the issue:**
1. Create a container using `docker run -dit --name ub1604_test s390x/ubuntu:16.04 /bin/bash`
2. Try to launch another bash session using `docker exec -it <container_id> /bin/bash`

**Describe the results you received:**
Got this error: `OCI runtime exec failed: exec failed: container_linux.go:348: starting container process caused "open /dev/pts/4294967296: no such file or directory": unknown`

**Describe the results you expected:**
Should be able to launch another bash session.

**Additional information you deem important (e.g. issue happens only occasionally):**
I uninstalled Docker CE v17.12.1 and install v17.09 but still saw the same error.  However if I then downgraded to v17.06 and I was able to create another bash session using `docker exec`.

**Output of `docker version`:**

```
# docker version
Client:
 Version:	17.12.1-ce
 API version:	1.35
 Go version:	go1.9.4
 Git commit:	7390fc6
 Built:	Tue Feb 27 22:17:54 2018
 OS/Arch:	linux/s390x

Server:
 Engine:
  Version:	17.12.1-ce
  API version:	1.35 (minimum version 1.12)
  Go version:	go1.9.4
  Git commit:	7390fc6
  Built:	Tue Feb 27 22:16:59 2018
  OS/Arch:	linux/s390x
  Experimental:	false
```

**Output of `docker info`:**

```
# docker info
Containers: 1
 Running: 1
 Paused: 0
 Stopped: 0
Images: 8
Server Version: 17.12.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 9b55aab90508bd389d7654c4baf173a981477d55
runc version: 9f9c96235cc97674e935002fc3d78361b696a69e
init version: 949e6fa
Security Options:
 apparmor
Kernel Version: 4.4.0-109-generic
Operating System: Ubuntu 16.04.1 LTS
OSType: linux
Architecture: s390x
CPUs: 8
Total Memory: 31.3GiB
Name: csz25089
ID: 3QQF:5HN3:HPS3:L25Z:EWGX:2LR4:HVPT:JY7T:NCHR:J72O:ILZQ:3Z5L
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by thaJeztah** [comment]
This looks like a bug in RunC 1.0.0-rc4 on "big-endian" architectures (which s390 is); https://github.com/opencontainers/runc/pull/1727

There's an open pull-request to update to RunC 1.0.0-rc5, which has this issue fixed; https://github.com/moby/moby/pull/36449

/cc @tophj-ibm

**Comment by thaJeztah** [comment]
Fix has been merged to master, and will be included in nightly builds of docker soon (e.g. https://download.docker.com/linux/ubuntu/dists/xenial/pool/nightly/amd64/). I also opened a ticket to request including the fix in the upcoming 18.03 Docker release (but it's up to the release team to decide if it will be included 😅 )

**Comment by vivkong** [comment]
Thanks!
